src/ptvsd/adapter/__main__.py

```diff
@@ -75,6 +76,15 @@ def _parse_argv():
         help="clear screen before starting the debuggee",
     )
 
+    parser.add_argument(
+        "--log-dir",
+        type=str,
+        nargs="?",
+        default=None,
+        metavar="DIR",
+        help="enable logging and use DIR to save adapter logs"
+    )
+
```

Comment:
If I'm reading this right, it would allow `--log-dir` without any command line argument, in which case it sets it to `None`, which disables it. I don't think that's what somebody who just said `--log-dir` would intend it to do.

So I think the argument should be mandatory - so no `nargs` and no `default`.